src/RoslynPad.Avalonia/DocumentViewFoldingManager.cs

```diff
@@ -0,0 +1,145 @@
+﻿using Avalonia.Threading;
```

Comment:
This code seems to be identical between Avalonia and WPF. I suggest you move all this code, including the stuff under the Folding dir to RoslynPad.Editor.Windows/Shared, which is also compiled in the RoslynPad.Editor.Avalonia project. You may need to add a few `#if`s to make it work, see other code in that project.

Then instantiate the folding manager inside RoslynCodeEditor instead of in DocumentView.

Reply:
Fixed